lib/paradox.js

```diff
@@ -17,418 +21,444 @@ var zoneToHASSLabel = {};
-					break;
-			}
-
-			this.ForwardVirtualZoneStatus(zone, status === "O");
-
-		}
-		else if (command.indexOf("RA") === 0) {
-			//RA001DOOOOOO
-			var status = command.substring(5, 6);
-		 	var alarmStatus = command.substring(10, 11);
-			var area = +command.substring(2, 5);
-			var state;
-			switch(status) {
-				case "D": case "F": case "I":
-					state = "disarmed";
-					break;
-				case "A":
-					state = "armed_away";
-					break;
-				case "S":
-					state = "armed_home";
-					break;
-			}
-			if (alarmStatus === "A")
-				state = "triggered";
-
-			this.SendMQTTEvent("paradox_evo/alarm/area/" + area, state);
-			console.log("Sending default state for area " + area + ": " + state + " to: " + "paradox_evo/alarm/area/" + area);
-
-		}
-		else if(command.indexOf("AA") === 0) {
-                        var area = +command.substring(2, 5);
-			if (command.split("&")[1] === "ok") {
-				console.log("Arm OK for Area: " + area);
-				this.SendMQTTEvent("paradox_evo/alarm/area/" + area, this._msg);
-			}
-		}
-                else if(command.indexOf("AD") === 0) {
-                        var area = +command.substring(2, 5);
-			if (command.split("&")[1] === "ok") {
-				console.log("Disarm OK for Area: " + area);
-                        	this.SendMQTTEvent("paradox_evo/alarm/area/" + area, "disarmed");
-			}
-                }
-		else if(command.indexOf("VC") === 0) {
-                        var zone = +command.substring(2, 5);
-                        if (command.split("&")[1] === "ok") {
-				//this.SendMQTTEvent("paradox_evo/alarm/virtual_zone/" + zone, "CLOSED");
-			}
+        return new Promise((resolve, reject) => {
+
+            if (!this._config.virtualZoneConfiguration) {
+                resolve();
+            }
+
+            var keys = Object.keys(this._config.virtualZoneConfiguration);
+
+            keys.forEach((key) => {
+                console.log("Subscribing: paradox_evo/alarm/virtual_zone/" + key + "/set");
+                this._mqttClient.subscribe("paradox_evo/alarm/virtual_zone/" + key + "/set");
+            });
+            resolve();
+        });
+
+    }
+
+    TranslateCommand(command) {
+
+        if (command.indexOf("G") === 0) {
+            this.ProcessSystemEvent(command);
+        } else if (command.indexOf("PGM") === 0) {
+            var pgmNum = +command.substring(3, 5);
+            var state = command.substring(5, 7) === "ON" ? "ON" : "OFF";
+
+            this.SendMQTTEvent("homeassistant/binary_sensor/paradox_vpgm" + pgmNum + "/state", state);
+        } else if (command.indexOf("ZL") === 0) {
+            var zone = +command.substring(2, 5);
+            var zoneConfiguration = this._config.zoneConfiguration;
+
+            if (zoneConfiguration[zone]) {
+                var label = command.substring(5, command.length).trim();
+                var nospc = (zoneConfiguration[zone].name ? zoneConfiguration[zone].name : label).replaceAll(" ", "_").unaccent().toLowerCase();
```

Comment:
Changed logic so `nospc` won't be read from alarm panel if configuration override is set.